include/seqan3/alignment/scoring/scoring_scheme_base.hpp

```diff
@@ -230,16 +233,24 @@ class scoring_scheme_base
      * \param[in] alph2   The second letter to score.
      * \return The score of the two letters in the current scheme.
      */
-    template <explicitly_convertible_to_concept<alphabet_t> alph1_t,
-              explicitly_convertible_to_concept<alphabet_t> alph2_t>
+    template <typename alph1_t,
+              typename alph2_t>
+    //!\cond
+        requires (explicitly_convertible_to_concept<alph1_t, alphabet_t> &&
+                  explicitly_convertible_to_concept<alph2_t, alphabet_t>)
+    //!\endcond
     constexpr score_t & score(alph1_t const alph1, alph2_t const alph2) noexcept
     {
         return matrix[to_rank(static_cast<alphabet_t>(alph1))][to_rank(static_cast<alphabet_t>(alph2))];
     }
 
     //!\copydoc score
-    template <explicitly_convertible_to_concept<alphabet_t> alph1_t,
-              explicitly_convertible_to_concept<alphabet_t> alph2_t>
+    template <typename alph1_t,
+              typename alph2_t>
+    //!\cond
+        requires (explicitly_convertible_to_concept<alph1_t, alphabet_t> &&
+                  explicitly_convertible_to_concept<alph2_t, alphabet_t>)
+    //!\endcond
```

Comment:
why these changes?

Reply:
I think you can discard all changes in this file or not?

Reply:
Good spot. I was just removing the char stuff. The only thing that will change is that I like at least our scoring schemes to expose the `alphabet_t` It is not required by the concept, but it makes some unit tests much easier to implement if I know what the alphabet of the scoring scheme will be. Is that ok for you?

Reply:
We could, but I'd prefer to mark it private. The thing is that the alphabet doesn't necessarily reflect what you are putting in, e.g. nucleotide_scoring_scheme always has alphabet_type dna15 although you can put all nucleotides in.

Reply:
But making it private is not really helpful. I don't see any danger coming with this. The type would be the underlying alphabet type. This information can be helpful and is basically custom stuff anyway, because the concept does not require it so within an algorithm we cannot and won't rely on it.

Reply:
Sure, but other people might rely on it and I think it would be better to not make something that is indeed implementation detail part of the API, after all the nucleotide_scoring_scheme might be changed to do something else internally, or not?

> But making it private is not really helpful.

You could make it private in the documentation maybe so you can still use it in tests?

But it's not that important. If you feel strongly about it, just add it :wink:

Reply:
I'll have a look in the evening. See if I can twerk around it.